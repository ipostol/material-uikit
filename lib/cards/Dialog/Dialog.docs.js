'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Dialog = require('./Dialog');

var _Dialog2 = _interopRequireDefault(_Dialog);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function () {
  return _react2.default.createElement(
    'div',
    { style: { height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' } },
    _react2.default.createElement(_Dialog2.default, {
      header: 'Delete design?',
      description: 'Deleting a design is permanent. There is no undo.',
      onClose: function onClose() {

        console.log('onClose');
      }
    })
  );
};