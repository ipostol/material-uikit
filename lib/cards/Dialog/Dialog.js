'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Close = require('../../icons/Close');

var _Close2 = _interopRequireDefault(_Close);

var _Button = require('../../buttons/Button');

var _Button2 = _interopRequireDefault(_Button);

var _Dialog = require('./Dialog.scss');

var _Dialog2 = _interopRequireDefault(_Dialog);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Dialog = function Dialog(_ref) {
  var header = _ref.header,
      description = _ref.description,
      onClose = _ref.onClose,
      primaryProps = _ref.primaryProps,
      secondaryProps = _ref.secondaryProps;
  return _react2.default.createElement(
    'div',
    { className: _Dialog2.default.general },
    _react2.default.createElement(
      'div',
      { className: _Dialog2.default.header },
      header
    ),
    _react2.default.createElement(
      'div',
      { className: _Dialog2.default.description },
      description
    ),
    _react2.default.createElement(
      'div',
      { className: _Dialog2.default.wrap },
      _react2.default.createElement(
        _Button2.default,
        _extends({
          mod: 'tertiary',
          className: _Dialog2.default.button,
          onClick: onClose
        }, secondaryProps),
        'Cancel'
      ),
      _react2.default.createElement(
        _Button2.default,
        _extends({
          mod: 'secondary',
          className: _Dialog2.default.button
        }, primaryProps),
        'Delete'
      )
    ),
    _react2.default.createElement(
      _Button2.default,
      { className: _Dialog2.default.close, onClick: onClose, mod: 'icon' },
      _react2.default.createElement(_Close2.default, null)
    )
  );
};

exports.default = Dialog;


Dialog.propTypes = {
  header: _propTypes2.default.string,
  description: _propTypes2.default.string,
  onClose: _propTypes2.default.func,
  primaryProps: _propTypes2.default.object,
  secondaryProps: _propTypes2.default.object
};