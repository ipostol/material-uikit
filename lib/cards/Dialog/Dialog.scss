@import '~bidstack-theme/theme';

.general {
  padding: 20px 17px 22px 24px;
  width: 448px;
  border-radius: 4px;
  background-color: $white3;
  box-shadow: 0 0 8px 0 $black5, 0 4px 8px 0 $black9;
  position: relative;
}

.header {
  font-size: $sizeHead2;
  line-height: 1.26;
  margin-bottom: 8px;
}

.description {
  font-size: $sizeBodyS;
  line-height: 1.5;
  margin-bottom: 16px;
}

.wrap {
  display: flex;
  justify-content: flex-end;
}

.button {
  height: 36px;
  width: 80px;
  margin-left: 8px;
}

.close {
  position: absolute;
  right: 8px;
  top: 8px;
}
