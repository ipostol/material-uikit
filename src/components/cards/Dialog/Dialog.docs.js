import React from 'react';
import Alert from './Dialog';

export default () => (
  <div style={{ height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
    <Alert
      header="Delete design?"
      description="Deleting a design is permanent. There is no undo."
      onClose={() => {

        console.log('onClose');

      }}
    />
  </div>
);
