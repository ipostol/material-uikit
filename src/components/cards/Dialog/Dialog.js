import React from 'react';
import PropTypes from 'prop-types';
import CloseIcon from '../../icons/Close';
import Button from '../../buttons/Button';
import styles from './Dialog.scss';

const Dialog = ({
  header,
  description,
  onClose,
  primaryProps,
  secondaryProps,
}) => (
  <div className={styles.general}>
    <div className={styles.header}>{header}</div>
    <div className={styles.description}>{description}</div>
    <div className={styles.wrap}>
      <Button
        mod="tertiary"
        className={styles.button}
        onClick={onClose}
        {...secondaryProps}
      >
        Cancel
      </Button>
      <Button
        mod="secondary"
        className={styles.button}
        {...primaryProps}
      >
        Delete
      </Button>
    </div>
    <Button className={styles.close} onClick={onClose} mod="icon">
      <CloseIcon />
    </Button>
  </div>
);

export default Dialog;

Dialog.propTypes = {
  header: PropTypes.string,
  description: PropTypes.string,
  onClose: PropTypes.func,
  primaryProps: PropTypes.object,
  secondaryProps: PropTypes.object,
};
