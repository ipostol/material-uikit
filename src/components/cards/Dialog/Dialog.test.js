import React from 'react';
import { shallow } from 'enzyme';
import Dialog from './Dialog';

describe('Dialog component', () => {

  it('should render correctly', () => {

    const props = {};

    const wrapper = shallow(
      <Dialog {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

});
