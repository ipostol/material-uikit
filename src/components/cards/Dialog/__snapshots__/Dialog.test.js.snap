// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component should render correctly 1`] = `
<div
  className="general"
>
  <div
    className="header"
  />
  <div
    className="description"
  />
  <div
    className="wrap"
  >
    <Button
      className="button"
      element="button"
      mod="tertiary"
      onClick={[Function]}
      round="small"
    >
      Cancel
    </Button>
    <Button
      className="button"
      element="button"
      mod="secondary"
      onClick={[Function]}
      round="small"
    >
      Delete
    </Button>
  </div>
  <Button
    className="close"
    element="button"
    mod="icon"
    onClick={[Function]}
    round="small"
  >
    <Close />
  </Button>
</div>
`;
